### PR TITLE
[RFC][ENH] OWWidget: Input/output summary 

### PR DESCRIPTION
**Pull request by ales-erjavec**
##### Issue

Provide a consistent widget input/output descriptions to the user.

##### Description of changes

Works on top of #2514

Add an input and output message areas in the widget's status bar.
The widget authors set the summaries using

```python
self.info.set_input_summary("foo")
self.info.set_output_summary("bar")
```

for basic inline short descriptions. A more detailed (including rich text) summaries are also possible:

screen shot 2017-08-31 at 13 24 12

<s>Ignore stylistic deficiencies in the last commit. Its flailing around to find some common 'description language'.</s>

##### Includes
- [X] Code changes
- [X] Tests
- [X] Documentation

### CODE REVIEW

**Comment by astaric** [comment]
@markotoplak, @BlazZupan, @janezd, this has been sitting here for too long.

If rebased, could it be included in the next release?

File: doc/development/source/tutorial-utilities.rst [hunk 1]
```diff
@@ -125,6 +125,59 @@ messags (`self.clear_message()`) apply to all messages of this type.
 `self.information(id, text)`, `self.warning(id, text)` and
 `self.error(id, text)` is deprecated and will be removed in the future.
 
+
+I/O Summaries
+-------------
+
+.. versionadded:: 3.7
```
Comment (markotoplak) [inline doc/development/source/tutorial-utilities.rst]: This line will probably have to updated. :)

**Comment by ajdapretnar** [comment]
This didn't make it into the 3.15.0 release, so where are we at?

**Comment by ajdapretnar** [comment]
@BlazZupan Can we merge this before 3.16 or at least decide what to do with it? @lanzagar

**Comment by BlazZupan** [comment]
This looks really cool, @ales-erjavec can you please resolve conflicts and then we merge.

File: doc/development/source/tutorial-utilities.rst [hunk 2]
```diff
@@ -125,6 +125,59 @@ messags (`self.clear_message()`) apply to all messages of this type.
...
+   self.info.set_output_summary("bar")
+
+Summaries are then displayed in the widget's status bar:
+
+.. image:: images/io-summary.png
+   :scale: 50 %
+   :alt: Inline status bar summary
+
+Predefined constants indicating no input/output are available as
+``self.info.NoInput`` and ``self.info.NoOutput`` respecively
```
Comment (janezd) [inline doc/development/source/tutorial-utilities.rst]: respec**t**ively

**Comment by BlazZupan** [comment]
Once merged, this functionality would also require a note in a widget development documentation. @ales-erjavec, can you please add an entry there as well.

**Pull request merged**